mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -177,35 +186,38 @@ def dialog_renderer(self):
         # Fall back to main language
         return self.dialog_renderers.get(self._core_lang)
 
+    @property
+    def settings_path(self):
+        if self.settings_write_path:
+            LOG.warning("self.settings_write_path has been deprecated!\n"
+                        "It goes against the XDG standards and causes confusion regarding settings location\n"
+                        "Support for overriding settings.json path will be removed in version 0.0.3")
+            return join(self.settings_write_path, 'settings.json')
+        return join(xdg.BaseDirectory.save_config_path(BASE_FOLDER, 'skills', self.skill_id), 'settings.json')
+
     def _init_settings(self):
         """Setup skill settings."""
+        if not self.skill_id:  # might not have been set yet when called in __init__ instead of bind
+            LOG.warning("_init_settings called before skill init completed")
+            return
 
-        # TODO remove this ugly ugly kludge
-        # To not break existing setups,
-        # save to skill directory if the file exists already
-        self.settings_write_path = Path(self.root_dir)
-
-        # Otherwise save to XDG_CONFIG_DIR
-        if not self.settings_write_path.joinpath('settings.json').exists():
-            self.settings_write_path = Path(xdg.BaseDirectory.save_config_path(
-                BASE_FOLDER, 'skills', basename(self.root_dir)))
-
-        # To not break existing setups,
-        # read from skill directory if the settings file exists there
-        settings_read_path = Path(self.root_dir)
-
-        # Then, check XDG_CONFIG_DIR
-        if not settings_read_path.joinpath('settings.json').exists():
-            for path in xdg.BaseDirectory.load_config_paths(BASE_FOLDER, 'skills',
-                                              basename(self.root_dir)):
-                path = Path(path)
-                # If there is a settings file here, use it
-                if path.joinpath('settings.json').exists():
-                    settings_read_path = path
-                    break
-
-        self.settings = get_local_settings(settings_read_path, self.name)
-        self._initial_settings = deepcopy(self.settings)
+        old_path = join(self.root_dir, 'settings.json')
+        # migrate settings to xdg if needed
+        # NOTE: mycroft-core setting have been migrated to XDG locations by default
+        # that is the reason we don't check for the XDG flag in ovos.conf
```

Comment:
will we maintain the xdg flag long term? if yes we should ensure it does what it says, otherwise we should remove the ovos.conf xdg flag completely once mycroft-core catches up

I'm inclined to maintain it for now and ensure it's respected, in that case should we use the old path when xdg is disabled?